Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -538,53 +538,54 @@ public enum TextfieldType: String {
             self.setNetworkActivityIndicator(status: true)
             // get public infor from server
             getPublicInfoFromServerJob.start(serverURL: self.serverURLNormalizer.normalizedURL, withCompletion: { (validatedURL: String?, _ serverAuthenticationMethods: Array<Any>?, _ error: Error?, _ httpStatusCode: NSInteger) in
-            
-                self.setNetworkActivityIndicator(status: false)
-                if (error != nil || validatedURL == nil) {
-                    
-                    self.setConnectButton(status: false)
-                    self.showURLError(
-                        self.manageNetworkErrors.returnErrorMessage(
-                            withHttpStatusCode: httpStatusCode, andError: error
+                
+                DispatchQueue.main.async {
```

Comment:
I suspect that you code this because you are changing the UI.
I'm not sure that all the code inside this main queue is needed to be here. you could add this inside the functions like `setConnectButton` and `showURLError` and only the code that changes the UI is going to be performed inside the main thread.  If you follow my suggestions you don't need to execute in the main thread the `getAuthMethodToLoginFrom`  and other things for example.

Reply:
In this case despite getAuthMethodToLoginFrom that is just a tiny static code, the rest update UI.